nipype/utils/misc.py

```diff
@@ -321,21 +325,28 @@ def dict_diff(dold, dnew, indent=0):
 
     diffkeys = len(diff)
 
+    def _shorten(value):
+        if isinstance(value, str) and len(value) > 50:
+            return f"{value[:10]}...{value[-10:]}"
+        if isinstance(value, (tuple, list)) and len(value) > 10:
+            return tuple(list(value[:2]) + "..." + list(value[-2:]))
+        return value
+
     # Values in common keys would differ quite often,
     # so we need to join the messages together
     for k in new_keys.intersection(old_keys):
-        try:
-            new, old = dnew[k], dold[k]
-            same = new == old
-            if not same:
-                # Since JSON does not discriminate between lists and
-                # tuples, we might need to cast them into the same type
-                # as the last resort.  And lets try to be more generic
-                same = old.__class__(new) == old
-        except Exception:
-            same = False
-        if not same:
-            diff += ["  * %s: %r != %r" % (k, dnew[k], dold[k])]
+        new = dnew[k]
+        old = dold[k]
+        # Reading from JSON produces lists, but internally we typically
+        # use tuples. At this point these dictionary values can be
+        # immutable (and therefore the preference for tuple).
+        if isinstance(dnew[k], (list, tuple)):
+            new = tuple([tuple(el) if isinstance(el, list) else el for el in dnew[k]])
+        if isinstance(dnew[k], (list, tuple)):
+            old = tuple([tuple(el) if isinstance(el, list) else el for el in dold[k]])
```

Comment:
What if there are dicts, or are nested beyond two levels? Should we just have a generic value uniformizer?

```Python
def uniformize(val):
    if isinstance(val, dict):
        return {k: uniformize(v) for k, v in val.items()}
    if isintsance(val, (list, tuple)):
        return tuple(uniformize(el) for el in val)
    return val
```

Then this is just:

```Python
new = uniformize(dnew[k])
old = uniformize(dold[k])
```